modules/MiniFigDirectionDetector.h

```diff
@@ -0,0 +1,59 @@
+/**
+ * @file   MiniFigDirectionDetector.h
+ * @brief  ミニフィグの向きを検出するクラス
+ * @author nishijima515
+ */
+
+#ifndef MINIFIG_DIRECTION_DETECTOR_H
+#define MINIFIG_DIRECTION_DETECTOR_H
+
+#include <opencv2/opencv.hpp>
+#include <opencv2/dnn.hpp>
+#include <algorithm>
+#include <iostream>
+#include <vector>
+#include <string>
+
+#define CONFIDENCE_THRESHOLD 0.4  // 検出結果を採用する最低信頼度の閾値
+#define NMS_THRESHOLD 0.4         // 検出ボックス同士の重なりを判断する閾値
+
+// ミニフィグの向きを表す列挙体
+enum class MiniFigDirection { FRONT, RIGHT, BACK, LEFT };
+
+struct MiniFigDirectionResult {
+  bool wasDetected = false;    // 検出が成功したかどうか
+  MiniFigDirection direction;  // ミニフィグの向きを表す列挙体
+};
+
+class MiniFigDirectionDetector {
+ public:
+  // コンストラクタ
+  MiniFigDirectionDetector(const std::string& modelPath = "etrobocon2025/yolo_optimized.onnx",
+                           const std::string& saveImagePath
```

Comment:
どういう画像を保存するか分かりにくいのと、その書き方だと動詞+目的語になってる。outputImagePathとかは？


あと、テストの時のために受け取るようにしてるって言ってたけど、datafiles/commandsみたいに、Makefileのtest-execのところにディレクトリを作るコマンドを実行するようにすれば、やっぱり引数で受け取る必要はないと思う。このクラス使うのも1走行で1回だけだと思うし、ない方がスッキリしてありがたい。
```suggestion
                           const std::string& outputImagePath
```